storey/drivers.py

```diff
@@ -501,3 +509,450 @@ async def _get_item(self, container, table_path, key, attributes):
+        """
+        The Redis key containing the associated timestamp for an aggregation.
+
+        The `redis_key_prefix` string should be the Redis key prefix used to store
+        data for a Storey container, table path, and key combination.
+        """
+        return f"{redis_key_prefix}:{self._aggregation_time_attribute_prefix}{feature_name}"
+
+    @staticmethod
+    def _list_key(redis_key_prefix, list_attribute_name):
+        """
+        The key containing a list of aggregation data.
+
+        The `redis_key_prefix` string should be the Redis key prefix used to store
+        data for a Storey container, table path, and key combination.
+        """
+        return f"{redis_key_prefix}:{list_attribute_name}"
+
+    @classmethod
+    def _convert_python_obj_to_redis_value(cls, value):
+        if isinstance(value, datetime):
+            if pd.isnull(value):
+                return None
+            return f'{cls.DATETIME_FIELD_PREFIX}{value.timestamp()}'
+        elif isinstance(value, timedelta):
+            return f'{cls.TIMEDELTA_FIELD_PREFIX}{value.total_seconds()}'
+        else:
+            if isinstance(value, float):
+                if value == math.inf or value == -math.inf:
+                    # Use the shorter infinity form (inf, -inf) to save space.
+                    value = f'{str(value)}'
+                elif not value % 1:
+                    # Store whole numbers as integers to save space.
+                    value = int(value)
+            return json.dumps(value)
+
+    def _convert_python_obj_to_lua_value(cls, value):
+        if isinstance(value, datetime):
+            if pd.isnull(value):
+                return None
+            return f'"{cls.DATETIME_FIELD_PREFIX}{value.timestamp()}"'
+        elif isinstance(value, timedelta):
+            return f'"{cls.TIMEDELTA_FIELD_PREFIX}{value.total_seconds()}"'
+        elif isinstance(value, bool):
+            return f'"{json.dumps(value)}"'
+        else:
+            if isinstance(value, float):
+                if value == math.inf:
+                    value = "math.huge"
+                elif value == -math.inf:
+                    value = "-math.huge"
+                elif not value % 1:
+                    # Store whole numbers as integers to save space.
+                    value = int(value)
+
+            return json.dumps(value)
+
+    @classmethod
+    def _convert_redis_value_to_python_obj(cls, value):
+        value = cls._convert_to_str(value)
+        if value.startswith(cls.TIMEDELTA_FIELD_PREFIX):
+            return timedelta(seconds=float(value.split(":")[1]))
+        elif value.startswith(cls.DATETIME_FIELD_PREFIX):
+            return datetime.fromtimestamp(float(value.split(":")[1]), tz=timezone.utc)
+        elif value == "inf":
+            return math.inf
+        elif value == "-inf":
+            return -math.inf
+        try:
+            return json.loads(value)
+        except Exception:
+            return value
```

Comment:
This could be pretty expensive, if it tries to parse each value as json and raises an exception (potentially for every row x column), only to return the original value. What redis types does this cover?

Reply:
To tell the truth I don't remember :). Should be tested without the exception

Reply:
fixed